ci/infra/testrunner/tests/conftest.py

```diff
@@ -12,6 +23,97 @@ def pytest_addoption(parser):
     parser.addoption("--vars", action="store",help="vars yaml" )
     parser.addoption("--platform", action="store",help="target platform" )
 
+
+class Workspace:
+    def __init__(self):
+        self.temporary_directory = tempfile.mkdtemp(prefix="skuba-tests")
+
+    def cleanup(self):
+        shutil.rmtree(self.temporary_directory)
+
+
+class ClusterFixture(contextlib.AbstractContextManager):
```

Comment:
I'm not sure using python's context has any advantage over using pytest fixture cleanup menchanis, which is simpler and more idiomatic to pytest.

Reply:
I am using https://docs.pytest.org/en/latest/fixture.html#factories-as-fixtures - mostly because it seems that otherwise tests will end up implementing the same fixtures again and again themselves like in the upgrade test: 

https://github.com/SUSE/skuba/blob/master/ci/infra/testrunner/tests/test_skuba_upgrade.py#L9-L15

Which is just a duplicate of: https://github.com/SUSE/skuba/blob/master/ci/infra/testrunner/tests/conftest.py#L26-L31

Because I don't see a way that a single test could easily say it needs a 3 master/3 worker cluster right now, if they have to use the configuration file values.

Reply:
Yes, Factory fixture is a great idea and definitely something other tests can use. my comment is about using abstract contentect manager for automatic tear down.